vpn/src/main/java/com/duckduckgo/mobile/android/vpn/ui/onboarding/DeviceShieldOnboardingActivity.kt

```diff
@@ -43,76 +40,57 @@ import com.duckduckgo.mobile.android.vpn.ui.onboarding.Command.RequestVPNPermiss
 
-    private lateinit var viewPager: ViewPager2
-
-    private lateinit var nextOnboardingPageCta: ImageButton
-    private lateinit var enableDeviceShieldLayout: View
-    private lateinit var onboardingFAQCta: Button
-    private lateinit var onboardingClose: ImageButton
-    private lateinit var enableDeviceShieldToggle: View
-
-    private inline fun <reified V : ViewModel> bindViewModel() = lazy { ViewModelProvider(this, viewModelFactory).get(V::class.java) }
-
     private val viewModel: DeviceShieldOnboardingViewModel by bindViewModel()
+    private val binding: ActivityDeviceShieldOnboardingBinding by viewBinding()
 
     @Suppress("DEPRECATION")
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        AndroidInjection.inject(this)
+        window.setStatusBarColor(getResources().getColor(com.duckduckgo.mobile.android.R.color.atp_onboardingHeaderBg))
+        if (isDarkThemeEnabled()) {
+            window.decorView.setSystemUiVisibility(View.SYSTEM_UI_FLAG_LIGHT_STATUS_BAR)
+        }
 
-        bindViews()
+        setContentView(binding.root)
         configureUI()
         observeViewModel()
     }
 
-    private fun bindViews() {
-        onboardingClose = findViewById(R.id.onboarding_close)
-        enableDeviceShieldToggle = findViewById(R.id.onboarding_switch_layout)
-        viewPager = findViewById(R.id.onboarding_pager)
-        enableDeviceShieldLayout = findViewById(R.id.onboarding_cta_layout)
-        onboardingFAQCta = findViewById(R.id.onboarding_faq_cta)
-        nextOnboardingPageCta = findViewById(R.id.onboarding_next_cta)
-    }
-
     private fun configureUI() {
-        viewPager.adapter = DeviceShieldOnboardingAdapter(viewModel.pages)
-        viewPager.registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
+        binding.onboardingPager.adapter = DeviceShieldOnboardingAdapter(viewModel.pages)
+        binding.onboardingPager.registerOnPageChangeCallback(object : ViewPager2.OnPageChangeCallback() {
 
             override fun onPageSelected(position: Int) {
                 showOnboardingPage(position)
                 super.onPageSelected(position)
             }
         })
 
-        onboardingFAQCta.setOnClickListener {
+        binding.onboardingFaqCta.setOnClickListener {
             DeviceShieldFAQActivity.intent(this).also {
                 startActivity(it)
             }
         }
 
-        nextOnboardingPageCta.setOnClickListener {
-            viewPager.currentItem = viewPager.currentItem + 1
+        binding.onboardingNextCta.setOnClickListener {
+            binding.onboardingPager.currentItem = binding.onboardingPager.currentItem + 1
         }
 
-        onboardingClose.setOnClickListener {
+        binding.onboardingClose.setOnClickListener {
             close()
         }
 
-        enableDeviceShieldToggle.setOnClickListener {
+        binding.onboardingSwitchLabel.setOnClickListener {
```

Comment:
When clicking "Enable App Tracking Protection", the toggle doesn't work. Shouldn't it be onboarding_switch_layout instead of onboardingSwitchLabel?